tests/test_generators.py

```diff
@@ -42,3 +43,20 @@ def test_double_generator_without_encoder():
     generated_double = EdmDouble.generate()
 
     assert generated_double == '1.2712595986497026e+39d'
+
+def test_string_generator_with_nonnegative():
+    random.seed(10)
+
+    mckString = StringNonNegativeMock(5,True)
+    generated_string = EdmString.generate(mckString)
+    
+    assert generated_string == "\'74894\'"
+
+def test_string_generator_without_nonnegative():
+    random.seed(10)
```

Comment:
I know why you choose the seed (10) for this test, but line comment will keep the knowledge for future generations :)

Reply:
I actually chose the seed completely arbitrarily, and kept it same for both non_negative tests :D

Reply:
But did it reproduced the bug before this fix?